### DISCUSSION TITLE
Please bring back MaintenanceModeException

### DISCUSSION THREAD

**Discussion by kristijorgji** [laravel/framework#41497] [Ideas]
Hi all, the discussion is here and closed without being fixed
https://github.com/laravel/framework/issues/35635

To summarize the issue is that `MaintenanceModeException` is not thrown anymore at laravel >=8.x and instead HttpClientException is thrown. So far so good who cares, if `$exception->getStatusCode() === 503` is maintenance mode we can handle this in our `Handler.php@report` method and return properly nice json if api or whatever we want.

**Now problem is:**
the previous exception MaintenanceModeException exposed very useful fields
`willBeAvailableAt` 
`wentDownAt`
`retryAfter`
 which are not available in HttpClientException

My request here is to bring back those fields either in HttpClientException somehow, or just return back MaintenanceModeException because in my opinion as developer makes total sense. Has concrete name of exception time and information that we use to nicely display user modal/ui with exact time of maitenance. 

Here is new exception with status code 503 replacing previous one. No information at all aside 503 status code
![Screenshot 2022-03-15 at 17 52 46](https://user-images.githubusercontent.com/15630854/158430287-46138341-adc3-4ed7-9610-ab6ae394014b.png)

In addition the `php artisan down` command now generates a file at  `storage/framework/maintenance.php` that is never called or used so looks redundant to me (was not created in pre 8.x versions. Before only json file storage/framework/down was created and used also storing the info of when again available etc ).

**Comment by willrowe** [comment]
The `storage/framework/maintenance.php` is referenced by the [`public/index.php` file](https://github.com/laravel/laravel/blob/660417c9f7d236bb702ca3d1f9e26d1764493e7a/public/index.php#L19), which is in the `laravel/laravel` repository.

**Reply by kristijorgji** [reply]
Ok seems like my upgrade from 7 to 8.4 went wrong probably my fault.
Because my index.php does not use that or even search for it

**Reply by kristijorgji** [reply]
Here is my public/index.php

```
<?php

/**
 * Laravel - A PHP Framework For Web Artisans
 *
 * @package  Laravel
 * @author   Taylor Otwell <taylor@laravel.com>
 */

/*
|--------------------------------------------------------------------------
| Register The Auto Loader
|--------------------------------------------------------------------------
|
| Composer provides a convenient, automatically generated class loader for
| our application. We just need to utilize it! We'll simply require it
| into the script here so that we don't have to worry about manual
| loading any of our classes later on. It feels nice to relax.
|
*/

require __DIR__.'/../bootstrap/autoload.php';

/*
|--------------------------------------------------------------------------
| Turn On The Lights
|--------------------------------------------------------------------------
|
| We need to illuminate PHP development, so let us turn on the lights.
| This bootstraps the framework and gets it ready for use, then it
| will load up this application so that we can run it and send
| the responses back to the browser and delight our users.
|
*/

$app = require_once __DIR__.'/../bootstrap/app.php';

/*
|--------------------------------------------------------------------------
| Run The Application
|--------------------------------------------------------------------------
|
| Once we have the application, we can handle the incoming request
| through the kernel, and send the associated response back to
| the client's browser allowing them to enjoy the creative
| and wonderful application we have prepared for them.
|
*/

$kernel = $app->make(Illuminate\Contracts\Http\Kernel::class);

$response = $kernel->handle(
    $request = Illuminate\Http\Request::capture()
);

$response->send();

$kernel->terminate($request, $response);

```

**Reply by kristijorgji** [reply]
I checked latest laravel index indeed checks for maintenance.php file. But why is it checking there while checking in middleware as well if down file exist.
Anyway my main question here is to return back those information and proper exception found it clearer and more helpful
